### PR TITLE
[JBPM-9797] Fix dispose engine for cancel dependent subprocess node

### PR DESCRIPTION
**Pull request by elguardian**
Jira: https://issues.redhat.com/browse/JBPM-9797

### CODE REVIEW

File: jbpm-flow/src/main/java/org/jbpm/workflow/instance/impl/WorkflowProcessInstanceImpl.java [hunk 1]
```diff
@@ -464,26 +463,26 @@ public void setState(final int state, String outcome) {
...
-                        } finally {
+                    RuntimeEngine runtime = null;
+                    try {
+                        runtime = manager.getRuntimeEngine(context);
+                        KieRuntime managedkruntime = runtime.getKieSession();
+                        managedkruntime.signalEvent("processInstanceCompleted:" + getId(), this);
+                    } catch (SessionNotFoundException e) {
+                        logger.debug("Could not found find parent process instance id {} for signaling completion", context.getContextId());
+                    } finally {
+                        if(manager != null) {
```
Comment (gmunozfe) [inline jbpm-flow/src/main/java/org/jbpm/workflow/instance/impl/WorkflowProcessInstanceImpl.java:481]: missing blank before parenthesis

File: jbpm-flow/src/main/java/org/jbpm/workflow/instance/node/SubProcessNodeInstance.java [hunk 2]
```diff
@@ -254,21 +254,27 @@ public void cancel(CancelType cancelType) {
...
-            		processInstance = (ProcessInstance) managedkruntime.getProcessInstance(processInstanceId);
-        	    } catch (SessionNotFoundException e) {
-        	        // in case no session is found for parent process let's skip signal for process instance completion
-        	    }
+                    processInstance = (ProcessInstance) managedkruntime.getProcessInstance(processInstanceId);
+                } catch (SessionNotFoundException e) {
+                    // in case no session is found for parent process let's skip signal for process instance completion
+                    logger.debug("Could not found find subprocess instance id {} for cancelling {}", context.getContextId(), cancelType);
+                } finally {
+                    if(runtime != null) {
```
Comment (gmunozfe) [inline jbpm-flow/src/main/java/org/jbpm/workflow/instance/node/SubProcessNodeInstance.java:274]: missing blank before parenthesis

**Review by gmunozfe — APPROVED** [review]
Looks good to me, fabulous work @elguardian !
Just a couple of minor formatting tips, in case you want to update them

**Comment by elguardian** [comment]
Jenkins retest this please

**Pull request merged**